sjsonnet/src-jvm-native/sjsonnet/SjsonnetMain.scala

```diff
@@ -291,4 +293,18 @@ object SjsonnetMain {
 
     }
   }
+
+  /**
+   * Read a path into a [[ResolvedFile]] if it exists and is a file. A resolved file acts as a layer
+   * of caching on top of the underlying file system. Small files are read into memory, while large
+   * files are read from disk.
+   */
+  private[this] def readPath(path: Path): Option[ResolvedFile] = {
+    val osPath = path.asInstanceOf[OsPath].p
+    if (os.exists(osPath) && os.isFile(osPath)) {
+      Some(new CachedResolvedFile(path.asInstanceOf[OsPath], memoryLimitBytes = 2048L * 1024L * 1024L))
```

Comment:
2GB seems a bit large for a threshold for reading files from disk. You mentioned being worried about 100mb-1gb JSON files hogging memory during parse, but given this threshold none of those would trigger the streaming parsing code path at all. 

Are there really that many >2gb jsonnet files in universe that streaming parsing just for those makes the difference? Or should this threshold be lower, e.g. 10mb? Or were all our tests done accidentally with streaming parsing mostly disabled, and actually we don't need it and reading each file fully into memory during the parse is fine?